pkg/apis/core/validation/cloudprofile.go

```diff
@@ -410,3 +412,99 @@ func validateRegions(regions []core.Region, fldPath *field.Path) field.ErrorList
 
 	return allErrs
 }
+
+func validateBastion(spec *core.CloudProfileSpec, fldPath *field.Path) field.ErrorList {
+	var (
+		allErrs     field.ErrorList
+		machineArch *string
+	)
+
+	if spec.Bastion == nil {
+		return allErrs
+	}
+
+	if spec.Bastion.MachineType != nil {
+		machineArch, allErrs = validateBastionMachineType(spec.Bastion.MachineType, spec.MachineTypes, fldPath.Child("machineType"))
+	}
+
+	if spec.Bastion.MachineImage != nil {
+		allErrs = append(allErrs, validateBastionImage(spec.Bastion.MachineImage, spec.MachineImages, machineArch, fldPath.Child("machineImage"))...)
+	}
+
+	return allErrs
+}
+
+func validateBastionMachineType(bastionMachine *core.BastionMachineType, machineTypes []core.MachineType, fldPath *field.Path) (*string, field.ErrorList) {
+	machineIndex := slices.IndexFunc(machineTypes, func(machine core.MachineType) bool {
+		return machine.Name == bastionMachine.Name
+	})
+
+	if machineIndex == -1 {
+		return nil, field.ErrorList{field.Invalid(fldPath.Child("name"), bastionMachine.Name, "machine type not found in spec.MachineTypes")}
+	}
+
+	return machineTypes[machineIndex].Architecture, nil
+}
+
+func validateBastionImage(bastionImage *core.BastionMachineImage, machineImages []core.MachineImage, machineArch *string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	namePath := fldPath.Child("name")
+
+	imageIndex := slices.IndexFunc(machineImages, func(image core.MachineImage) bool {
+		return image.Name == bastionImage.Name
+	})
+
+	if imageIndex == -1 {
+		return append(allErrs, field.Invalid(namePath, bastionImage.Name, "image name not found in spec.MachineImages"))
+	}
+
+	imageVersions := machineImages[imageIndex].Versions
+
+	if bastionImage.Version == nil {
+		allErrs = append(allErrs, checkImageSupport(bastionImage.Name, imageVersions, machineArch, namePath, core.ClassificationSupported)...)
+	} else {
+		versionPath := fldPath.Child("version")
+
+		versionIndex := slices.IndexFunc(imageVersions, func(version core.MachineImageVersion) bool {
+			return version.Version == *bastionImage.Version
+		})
+
+		if versionIndex == -1 {
+			return append(allErrs, field.Invalid(versionPath, bastionImage.Version, "image version not found in spec.MachineImages"))
+		}
+
+		validClassifications := []core.VersionClassification{
+			core.ClassificationSupported,
+			core.ClassificationPreview,
+		}
```

Comment:
Can you explain this - why do we allow 'supported' and 'preview' here, while we only allow 'supported' when no version is given?

Reply:
If I remember correctly we had the case in the past that all gardenlinux versions which were supported contained a critical CVE and only the ones in preview did not.
For such cases we could potentially define the version to an image which is only in preview.
But on second thought this might be an utopian use case.
@kon-angelo what do you think?

Reply:
Yes, it sounds a bit weird. In such case (all `supported` versions have CVE), shoots are also affected, so the operator is probably very much interested in promoting a new version to `supported` (that might have been in `preview` before). I vote for removing this complexity here.

Reply:
Had a quick discussion with @kon-angelo and we agreed to only allow supported images.
Changed the logic accordingly with the latest commit.